pkgs/applications/science/misc/root/default.nix

```diff
@@ -28,18 +28,27 @@ stdenv.mkDerivation rec {
     substituteInPlace cmake/modules/SearchInstalledSoftware.cmake \
       --replace 'set(lcgpackages ' '#set(lcgpackages '
 
+    # Don't require textutil on macOS
+    : > cmake/modules/RootCPack.cmake
+
+    # Hardcode path to fix use with cmake
+    sed -i cmake/scripts/ROOTConfig.cmake.in \
+      -e 'iset(nlohmann_json_DIR "${nlohmann_json}/lib/cmake/nlohmann_json/")'
+
     patchShebangs build/unix/
   '' + lib.optionalString noSplash ''
     substituteInPlace rootx/src/rootx.cxx --replace "gNoLogo = false" "gNoLogo = true"
   '';
 
   cmakeFlags = [
     "-Drpath=ON"
+    "-DCMAKE_CXX_STANDARD=17"
```

Comment:
Why was this needed? In c++11 and c++14 code this can cause errors like
```
In file included from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:29,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TCollection.h:29,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TSeqCollection.h:25,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TObjArray.h:25,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TClonesArray.h:25,
                 from ../StSpinPool/StJetEvent/StJetEvent.cxx:7:
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/ROOT/RStringView.hxx:84:17: error: expected type-specifier
   84 |        operator std::string_view() const { return std::string_view(fData,fLength); }
      |                 ^~~
In file included from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TCollection.h:29,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TSeqCollection.h:25,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TObjArray.h:25,
                 from /nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TClonesArray.h:25,
                 from ../StSpinPool/StJetEvent/StJetEvent.cxx:7:
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:115:13: error: expected type-specifier
  115 |    operator std::string_view() const { return std::string_view(Data(),fExtent); }
      |             ^~~
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:280:32: error: ‘string_view’ in namespace ‘std’ does not name a type
  280 |    explicit TString(const std::string_view &sub);
      |                                ^~~~~~~~~~~
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:280:27: note: ‘std::string_view’ is only available from C++17 onwards
  280 |    explicit TString(const std::string_view &sub);
      |                           ^~~
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:317:37: error: ‘string_view’ in namespace ‘std’ does not name a type
  317 |    TString    &operator=(const std::string_view &s);
      |                                     ^~~~~~~~~~~
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:317:32: note: ‘std::string_view’ is only available from C++17 onwards
  317 |    TString    &operator=(const std::string_view &s);
      |                                ^~~
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:444:9: error: ‘string_view’ in namespace ‘std’ does not name a type
  444 |    std::string_view View() const { return std::string_view(GetPointer(),Length()); }
      |         ^~~~~~~~~~~
/nix/store/rp2sd4kxrzgq22913q8qqag5b2i8j48z-root-6.24.06/include/TString.h:444:4: note: ‘std::string_view’ is only available from C++17 onwards
  444 |    std::string_view View() const { return std::string_view(GetPointer(),Length()); }
      |    ^~~
```

Reply:
I don't remember, and I think this is probably NOT needed. Sorry. 

The most likely scenario is: I enabled C++ 17 in my local ROOT version, and that worked fine so I didn't change that flag back in the pull request, which is based on my local ROOT.